src/Datadog.Trace/Configuration/StringConfigurationSource.cs

```diff
@@ -15,8 +15,9 @@ public abstract class StringConfigurationSource : IConfigurationSource
         /// <paramref name="data"/>.
         /// </summary>
         /// <param name="data">A string containing key-value pairs which are comma-separated, and for which the key and value are colon-separated.</param>
+        /// <param name="allowOptionalMappings">Determines whether to create dictionary entries when the input has no value mapping</param>
         /// <returns><see cref="IDictionary{TKey, TValue}"/> of key value pairs.</returns>
-        public static IDictionary<string, string> ParseCustomKeyValues(string data)
+        public static IDictionary<string, string> ParseCustomKeyValues(string data, bool allowOptionalMappings = false)
```

Comment:
Not sure if it's an issue, but this is a breaking change in a public API. Maybe we should add as an overload without optional parameter instead and delegate the existing call to this one?

Reply:
Yes you're absolutely right, I'll make that change 👍🏼

Reply:
Done in b3f9c8c9975f0bbf1ada23a363a69edf4e77ac8d